test/test_foreach.py

```diff
@@ -54,7 +54,11 @@ def _test_bin_op_list(self, device, dtype, foreach_op, foreach_op_, torch_op):
             tensors1 = self._get_test_data(device, dtype, N)
             tensors2 = self._get_test_data(device, dtype, N)
 
-            expected = [torch_op(tensors1[i], tensors2[i]) for i in range(N)]
+            # Mimics cuda kernel dtype flow.  With fp16/bf16 input, runs in fp32 and casts output back to fp16/bf16.
+            control_dtype = torch.float32 if (device == "cuda" and
+                                              (torch.float16 or dtype is torch.bfloat16)) else dtype
+            expected = [torch_op(tensors1[i].to(dtype=control_dtype),
```

Comment:
you just caught my bug. i was wondering how can this work and just realized that `expected` is not being used.